src/imageio/storage/piwigo.c

```diff
@@ -1040,18 +1084,37 @@ int store(dt_imageio_module_storage_t *self, dt_imageio_module_data_t *sdata, co
 
     if(status)
     {
-      status = _piwigo_api_upload_photo(p, fname, author, caption, description);
-      if(!status)
+      char *pwg_image_id = _piwigo_api_get_image_id(p, img);
+
+      if(dt_bauhaus_combobox_get(ui->conflict_action)==2)
```

Comment:
you cannot Gtk GUI routine in store(). All Gtk routines must be used on a single thread initiated at dt startup.

The conflict_action status must be stored into the parameters, see `get_param()`.